src/index.js

```diff
@@ -88,9 +88,18 @@ const ChartComponent = React.createClass({
 		nextData.forEach(function (dataset, sid) {
 			if (currentData[sid] && currentData[sid].data) {
 				currentData[sid].data.splice(nextData[sid].data.length);
+
 				dataset.data.forEach(function (point, pid) {
 					currentData[sid].data[pid] = nextData[sid].data[pid];
 				});
+
+				delete dataset.data;
```

Comment:
hmm.. sorry its early here. :-p Do you mean a complete line replacement of delete to your proposal like this?

```
nextData.forEach(function (dataset, sid) {
			if (currentData[sid] && currentData[sid].data) {
				currentData[sid].data.splice(nextData[sid].data.length);

				dataset.data.forEach(function (point, pid) {
					currentData[sid].data[pid] = nextData[sid].data[pid];
				});

				const { data, ...dataset } = dataset;

				currentData[sid] = {
					data: currentData[sid].data,
					...currentData[sid],
					...dataset
				}
			} else {
				currentData[sid] = nextData[sid];
			}
		});
```